include/fluent-bit/flb_config.h

```diff
@@ -154,6 +154,16 @@ struct flb_config {
      */
     char *http_proxy;
 
+    /*
+     * A comma-separated list of host names that shouldn't go through
+     * any proxy is set in (only an asterisk, * matches all hosts).
+     * As a convention (https://curl.se/docs/manual.html), this value can be set
+     * and respected by `NO_PROXY` environment variable when `HTTP_PROXY` is used.
+     * Example: NO_PROXY="127.0.0.1,localhost,kubernetes.default.svc"
+     * Note: only `,` is allowed as seperator between URLs.
+     */;
+    char *no_proxy;
```

Comment:
Can we support IP ranges in CIDR format? Check https://docs.openshift.com/enterprise/3.2/install_config/http_proxies.html?

Reply:
I don't think this is feasible for now. This is because that we only have `hostname` when creating a `http_upstream`. I don't know a easy / feasible way for turning that into an IP address and do the CIDR check. 

(Golang supports that natively. Not so easy with the fluent-bit's C code base. )

If you have some ideas, feel free to enhance this feature based on my PR.

Reply:
I think if IP address was configured in config file, the value of hostname should be IP address which you check against with CIDR.

Reply:
The hostname is something uncertain, depends on the remote host. For example, for stackdriver output plugin it will be the `logging.googleapis.com`. Almost certainly we will have a domain name instead of a IP address here, and since the whole fluent-bit's http client is implemented in its own codebase, doing DNS lookup etc. here would be non-trivial. 

So again, though this is nice to have, I would consider this as a separate feature requirement. Feel free to contribute PR for this.